### PR TITLE
Missing `VirtualFile.run` usage in `DirectoryBrowserSupport`

### PR DESCRIPTION
**Pull request by jglick**
I found a failing test in `artifact-manager-s3` during a routine dependency bump including a core version change, and bisected the failure to a security fix 80452662b31ac6c9f4418cffae1af6af4daf479a. This neglected to use `VirtualFile.run` to batch together file metadata operations, causing the number of S3 API calls in the test case to jump from 13 to 37. A simple reorganization of method calls restores the previous behavior so far as this test case is concerned.

### Testing done

https://github.com/jenkinsci/artifact-manager-s3-plugin/pull/450#issuecomment-1896022561

### Proposed changelog entries

- A security fix in 2.394 caused a substantial slowdown in displaying build artifacts when using remote artifact managers such as in S3.

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by daniel-beck — APPROVED** [review]
Looks reasonable, thanks! Haven't manually tested.

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by MarkEWaite — APPROVED** [review]

**Comment by MarkEWaite** [comment]
Included in [Jenkins 2.443](https://www.jenkins.io/changelog/?query#v2.443)

**Pull request merged**
